plugins/catalog-backend/src/processing/DefaultCatalogProcessingEngine.ts

```diff
@@ -129,9 +129,13 @@ export class DefaultCatalogProcessingEngine {
   }
 
   private startPipeline(): () => void {
+    const lowWatermark =
+      this.config.getOptionalNumber('catalog.pipeline.lowWatermark') ?? 5;
+    const highWatermark =
+      this.config.getOptionalNumber('catalog.pipeline.highWatermark') ?? 10;
```

Comment:
Should throw an exception if min isn't strictly smaller than max.

I'm also not sure, should you really be permitted to define just one of them? That'll just cause problems if we want to change the defaults - then people's installations may start to break if that violates this check. So, maybe `concurrency` should be optional, but not `min` and `max` inside it?